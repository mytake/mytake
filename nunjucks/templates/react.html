{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
