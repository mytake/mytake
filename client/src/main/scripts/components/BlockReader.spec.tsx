/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2018 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as React from "react";
import * as renderer from "react-test-renderer";
import BlockReader from "./BlockReader";
import { TakeDocument } from "./BlockEditor";

const doc: TakeDocument = {
  title: "My take title",
  blocks: [
    {
      kind: "paragraph",
      text: "",
    },
  ],
};

jest.mock("./ReactionContainer", () => ({
  __esModule: true,
  default: "ReactionContainer",
}));

function createNodeMock(element: React.ReactElement<HTMLElement>) {
  switch (element.type) {
    case "div": {
      return {
        resetHeight() {
          // Do nothing
        },
      };
    }
    case "textarea": {
      return {
        focus() {
          // Do nothing
        },
      };
    }
    case "p": {
      return {
        resetHeight() {
          // Do nothing
        },
      };
    }
    case "h1": {
      return {
        focus() {
          // Do nothing
        },
      };
    }
  }
}

test("Read only", () => {
  // Object.defineProperty(window.location, 'pathname', {
  // 	writable: true,
  // 	value: 'anything'
  // });

  const tree = renderer
    .create(<BlockReader initState={doc} takeId={2} />, { createNodeMock })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
