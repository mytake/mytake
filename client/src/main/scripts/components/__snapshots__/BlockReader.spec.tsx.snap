// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read only 1`] = `
<div>
  <div
    className="editor__wrapper"
  >
    <div
      className="editor"
    >
      <div
        className="editor__inner"
      >
        <div
          className="editor__text-container"
        >
          <h1
            className="editor__title"
            style={
              Object {
                "height": undefined,
              }
            }
          >
            My take title
          </h1>
          <div
            className="editor__title-height-div"
          />
        </div>
        <div
          className="editor__block-list"
        >
          <div
            className="editor__block editor__block--base"
          >
            <div
              className="editor__text-container"
            >
              <p
                className="editor__paragraph"
                style={
                  Object {
                    "height": undefined,
                  }
                }
              >
                
              </p>
              <div
                className="editor__paragraph-height-div"
              >
                <p>
                  <br />
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <ReactionContainer
    takeDocument={
      Object {
        "blocks": Array [
          Object {
            "kind": "paragraph",
            "text": "",
          },
        ],
        "title": "My take title",
      }
    }
    takeId={2}
  />
</div>
`;
